core/src/main/scala/akka/persistence/cassandra/journal/TagWriters.scala

```diff
@@ -295,40 +310,34 @@ import akka.util.ByteString
         val startTime = System.nanoTime()
 
         def writeTagScanningBatch(group: Seq[(String, Long)]): Future[Done] = {
-          val batch = new BatchStatement(BatchStatement.Type.UNLOGGED)
-          group.foreach {
-            case (pid, seqNr) => batch.add(ps.bind(pid, seqNr: JLong))
+          val statements: Seq[BoundStatement] = group.map {
+            case (pid, seqNr) => ps.bind(pid, seqNr: JLong)
           }
-          tagWriterSession.executeStatement(batch)
+          Future.traverse(statements)(tagWriterSession.executeStatement).map(_ => Done)
         }
 
-        // Group the updates into 500 statements per UNLOGGED BatchStatement. These
-        // are executed sequentially to not induce too much load that might influence
-        // performance of other writes and reads. See issue #408.
-        // The size of the data is small and fixed so no need to configure the batch size.
-        val batchIterator: Iterator[Future[Done]] =
-          updates.grouped(500).map(writeTagScanningBatch)
-
-        def next(): Future[Done] =
-          if (batchIterator.hasNext) batchIterator.next().flatMap(_ => next())
-          else Future.successful(Done)
-
-        val result: Future[Done] = next()
-
-        result.onComplete {
-          case Success(_) =>
-            if (log.isDebugEnabled)
-              log.debug(
-                "Update tag scanning of [{}] pids took [{}] ms",
-                updates.size,
-                (System.nanoTime() - startTime) / 1000 / 1000)
-          case Failure(t) =>
-            log.warning("Writing tag scanning failed. Reason {}", t)
-            self ! TagWriteFailed(t)
+        // Execute 10 async statements at a time to not introduce too much load see issue #408.
+        val batchIterator: Iterator[Seq[(PersistenceId, SequenceNr)]] = updates.grouped(10)
+
+        var result = Future.successful[Done](Done)
+        for (item <- batchIterator) {
+          result = result.flatMap { _ =>
+            writeTagScanningBatch(item)
+          }
+        }
+
+        result.onComplete { result =>
+          self ! WriteTagScanningCompleted(result, startTime, updates.size)
+          result.failed.foreach(self ! TagWriteFailed(_))
```

Comment:
@chbatey please review this part once more, originally I missed the `TagWriteFailed` send to self here